test/test_execution.py

```diff
@@ -193,3 +196,29 @@ def test_execute_unknown_command(
     assert execute(args) == 127
     assert f"{cmd}: command not found\n" == capsys.readouterr().err
     assert_expected_files(temp_output_dir, exists=False)
+
+
+def test_signal_exit(temp_output_dir: str) -> None:
+
+    def runner() -> int:
+        args = Arguments.from_argv(
+            ["sleep", "60"],
+            output_prefix=temp_output_dir,
+        )
+        return execute(args)
+
+    thread = threading.Thread(target=runner)
+    thread.start()
+    sleep(0.01)  # make sure the process is started
+    ps_command = "ps aux | grep '[s]leep 60'"  # brackets to not match grep process
```

Comment:
fragile since could be some other process on accasion running some `sleep 60` etc.  since actual duration does not matter, make it larger and more likely to be "unique", e.g. 6903123